### ISSUE TITLE
webfonts are no longer bundled when using StyleExtHtmlWebpackPlugin

### ISSUE THREAD

**Issue by firsttris** [numical/style-ext-html-webpack-plugin#7]
Hello,

i want to replace external css with internal css (style tag).

i tried to replace ExtractTextPlugin with StyleExtHtmlWebpackPlugin.

In general its working but there is one issue:
my webfonts (woff|eot|ttf|otf|svg) are no longer bundled into the dist directory, and i have no clue why.

I uploaded my webpack config:

Folder-1: webpack config with ExtractTextPlugin and external CSS (working)
Folder-2: webpack config with StyleExtHtmlWebpackPlugin and internal CSS (missing webfonts)

https://drive.google.com/drive/folders/0B7mzI0TL-AbULUdhQ2tpLW5vR3M?usp=sharing

maybe you can help or point into the right direction?

best regards
tristan

**Comment by numical** [comment]
Thanks for the heads-up.
I will take a look this week.

**Comment by numical** [comment]
Progress update:
Having first had to update test for the latest webpack 2 beta, I finally got around to looking at this.
I have a [test that replicates the problem](https://github.com/numical/style-ext-html-webpack-plugin/blob/webfont-issue/spec/web-font-spec.js) on branch 'webfont-issue'.
Now to what to do about it...
Caution: StyleExtHtmlWepbackPlugin is an awful lot less clever then ExtractTextWebpackPlugin so don't expect a quick fix, sorry,

**Comment by numical** [comment]
**tl;dr**
Use a hack until I get this fixed: add an explicit `require` for each of the font files you need; ie:
```javascript
require ('mystyles.css');
require('myfont.woff');
require('myfont.woff2');
\\ etc.
```

Ok - the issue is that StyleExtHtmlWepbackPlugin does not use (and in fact [is incompatible with](https://github.com/numical/style-ext-html-webpack-plugin#basic-usage)) [css-loader](https://github.com/webpack/css-loader) and it is css-loader that ensures the local font files referenced in `@font-face` declarations are included in the build.  StyleExtHtmlWepbackPlugin needs to replicate this behaviour.

**Comment by firsttris** [comment]
hey thx for your reply ! this hack would work indeed.

since im using .css and .styl (stylus) files im using the following configuration for ExtractTextPlugin to extract everything in a .css file. 

  vueConfig.loaders = {
    stylus: ExtractTextPlugin.extract({
      test: /\.styl/,
      loader: 'css-loader!stylus-loader',
      fallbackLoader: 'vue-style-loader' // <- this is a dep of vue-loader
    }),
    css: ExtractTextPlugin.extract({
      test: /\.css/,
      loader: 'css-loader',
      fallbackLoader: "vue-style-loader"
    })
  }
 
and later:
new ExtractTextPlugin('css/styles.[hash].css'),

i would like todo exacly the same but generate internal css (<style> tag)

actually i thought this would be possible with ExtractTextPlugin, but it doesn't seem so...

but - would it be possible that your plugins just consumes the String extracted by ExtractTextPlugin and create Internal <style> tag?

regards
Tristan

**Comment by numical** [comment]
Note you will also loose the ability to have `[hash]` in the names; the font files will be renamed properly but the inlined css will not have the hash added.

As for simply using the ExtractTextPlugin output - you're not the first to suggest it!  However two gotcha's:
- getting the sequencing right during the webpack emit event so that I can read (and remove the ExtractTextPlugin content) before the HTMLWebpackPlugin content is written;
- design-wise it is plain ugly: explicitly add an 'Extract' plugin, then delete its contents and move them elsewhere - it smells don't you think?

**Comment by firsttris** [comment]
sorry i have absolute no inside knowledge in those webpack plugins. i was just looking at it from a user perspective :) just tried to explain what i wanted to achieve. 

but i understand now its not that easily possible to integrate 3 plugins. So your plugin is doing somehow the same job that ExtractTextPlugin  is doing and than generating Internal CSS.

just to clarify: 
on the readme your are saying "generating inline css". 
but when i used your plugin it was generating "internal css". 

inline css:
<div style="color:red;">

vs.

internal css
<style>
.div {
color: red;
}
</style>

vs.

external

reference:
http://www.w3schools.com/css/css_howto.asp
http://stackoverflow.com/questions/20288218/what-is-difference-between-external-and-internal-css

im a bit confused?
is it really possible to generate "inline css"?

**Comment by numical** [comment]
Sorry for the delayed reply.
No problem with you expressing what you want!
In fact you inspired me to try using the ExtractTextWebpackPlugin output, and it turns out all my fears about scheduling issues were unfounded - it has proved to be fairly simple (...so far..).
The result is my [v3 beta](  https://github.com/numical/style-ext-html-webpack-plugin/tree/v3-beta) which should be useable this week.  I'll let you know...

As for 'inline' vs 'internal' css - I'm a typical native English speaker being sloppy - I mean 'internal'.  Will get the README fixed too.

**Comment by firsttris** [comment]
looking forward to test your newest version 👍

**Comment by numical** [comment]
v3.0.0 just released - working as you suggested ;-) 
Note it has different but simpler configuration than v2 (no loader).
Let me know if you have any issues.

**Comment by firsttris** [comment]
im eager to test :)
but npm still installs 2.0.6
can you push v3.0.0 to npm?
thx!
